request_test.go

```diff
@@ -7,18 +7,20 @@ import (
 	"testing"
 )
 
-var ignoreSpacesRe = regexp.MustCompile(">[ \t\r\n]+<")
+// match leading and trailing whitespace on each line
+var ignoreSpacesRe = regexp.MustCompile("(?m)^[ \t]+|$[\r\n]+")
```

Comment:
Does this remove non-line-ending whitespace (i.e. ` ` and `\t`) from the ends of lines, or only from the beginning? I confess I haven't done much with multi-line regexes, so I may be missing something.

Reply:
Ah, you're right! I did miss the trailing whitespace. I guess it didn't show up in tests since typically the lines didn't have trailing spaces. Fixed in bd83dc7